### ISSUE TITLE
MC Behind Nginx Proxy Manager 

### ISSUE THREAD

**Issue by antjrobles** [Ylianst/MeshCentral#3160]
Hi, Im trying to run MC behind nginx proxy manager but I cant make it, I have tried many things, read many posts and follow guides, but without any results. This is my config.json. Any ideas? Thanks

"TLSOffload": "139.162.166.214". This is the IP where Nginx proxy manager is. Thanks again

> {
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "settings": {
    "cert": "mesh.antolinode.tk",
    "WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 4430,
    "aliasPort": 443,
    "redirPort": 80,
    "redirAliasPort": 80,
    "AgentPong": 300,
    "TLSOffload": "139.162.166.214",
    "BrowserPong": 20,
    "SelfUpdate": false,
    "AllowFraming": "false",
    "WebRTC": "true"
  },
  "domains": {
	"": {
	"title": "MyServer",
    "_title2": "Servername",
    "_minify": true,
    "NewAccounts": "false",
	"_userNameIsEmail": true,
    "certUrl": "https://mesh.antolinode.tk:443/"
	}
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before>",
    "_email": "myemail@mydomain.com",
    "_names": "myserver.mydomain.com",
	"production": false
  }
}

**Comment by nzalev** [comment]
I noticed that you are using TLS offload with a public IP. Are you proxying over the internet between two separate servers?

When using TLS offloading, Meshcentral will use HTTP and not HTTPS, so if you are proxying to https://meshcentral.yourdomain.com, this will fail with an SSL protocol error.

If you are proxying between two separate servers and intending on using HTTPS between them, change "TLSOffload" to "TrustedProxy". With this option, meshcentral will continue to use HTTPS.

If you are using a reverse proxy on the same server instance as meshcentral, you may need to change the IP to "127.0.0.1".  If you are going to be using TLS Offloading, ensure you are proxying to http:// and not https://.

**Comment by antjrobles** [comment]
Thanks for your reply. I´ll try and let you know as soon as possible. Im busy these days. By the way, as you can see Im new in this, im learning. Thanks again.

**Comment by antjrobles** [comment]
And yes, im trying proxing over the internet between two separates server. The meshcentral docker is installed in a VPS in Linode, and trying to connect to my server, that is at at home.

**Comment by sovcali** [comment]
I have a fully working MeshCentral with NPM and Docker, in my case I set "certUrl" with the same "local-container" IP of NGINX, but not sure if thats the solution for you, I'm currently working in making a repository here with my config files but I'm really new on Github

**Comment by ONBET** [comment]
no update?

**Comment by poiNt3D** [comment]
I have spent much time trying everything,
but failed to notice that
"certUrl" in my config
is actually  "_certUrl".
After fixing
worked like a charm.

**Comment by nzalev** [comment]
> I have spent much time trying everything, but failed to notice that "certUrl" in my config is actually "_certUrl". After fixing worked like a charm.

Yes, if you are runnning meshcentral behind an SSL terminating proxy, meshcentral needs to know where to find the certificate through the `certurl` domain option.

**Comment by JSkolnik** [comment]
Meshcentral for NPM works.But I don't work Intel AMT (Cira).I don't know how to set up NPM.

**Comment by si458** [comment]
closing as stale.
please try the latest version 1.1.21 and node 18 or above,
if issue still persists, please reply back
